### PR TITLE
Parallelize wheel-builds where applicable

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:**
This PR will parallelize wheel-builds

**Description of the Change:**
* Wheel-builds parallelized where applicable.

**Benefits:**
* Faster wheel-builds time.

**Possible Drawbacks:**
Not that I'm aware of.

**Related GitHub Issues:**

### CODE REVIEW

**Comment by AmintorDusko** [comment]
Pull request events are triggering all wheels workflows, and artifacts uploaded, for tests.
This will be reverted before merging.

**Comment by AmintorDusko** [comment]
Benchmarks:
  | Master | Parallel(Py version)
-- | -- | --
Wheel::Linux::x86_64 | 34m 5s | 12m 30s
Wheel::MacOS::Intel | 39m 34s | 11m 44s
Wheel::Linux::ARM | 6h 0m 0s (time out) | 1h 56m
Wheel::Windows::x86_64 | 14m 12s |  
Wheel::MacOS::ARM | 27m 15s | 15m 3s
Wheel::Linux::PowerPC | 5h 45m 55s | 1h 26m

File: .github/workflows/wheel_linux_aarch64.yml
Comment (mlxd) [inline .github/workflows/wheel_linux_aarch64.yml:3]: Should we remove this?

File: .github/workflows/wheel_linux_aarch64.yml
Comment (mlxd) [inline .github/workflows/wheel_linux_aarch64.yml:65]: Swap these?

File: .github/workflows/wheel_linux_ppc64le.yml
Comment (mlxd) [inline .github/workflows/wheel_linux_ppc64le.yml:3]: As above?

File: .github/workflows/wheel_linux_ppc64le.yml
Comment (mlxd) [inline .github/workflows/wheel_linux_ppc64le.yml:62]: Ditto

File: .github/workflows/wheel_macos_arm64.yml
Comment (mlxd) [inline .github/workflows/wheel_macos_arm64.yml:22]: We should probably double-check the wheels from this on @isschoch or @multiphaseCFD Macs before merging.

**Review by mlxd — COMMENTED** [review]
Nothing but some tidy-up suggestions for when we are ready. Great work putting this together! @multiphaseCFD & @isschoch can you both touch-base with @AmintorDusko to help check the MacOS wheels locally when you have time?

File: .github/workflows/wheel_macos_arm64.yml
Comment (AmintorDusko) [reply .github/workflows/wheel_macos_arm64.yml:22]: I will wait for their feedback.

File: .github/workflows/wheel_linux_aarch64.yml
Comment (multiphaseCFD) [reply .github/workflows/wheel_linux_aarch64.yml:3]: will try it later and get back to you guys. @mlxd @AmintorDusko

**Comment by multiphaseCFD** [comment]
> Nothing but some tidy-up suggestions for when we are ready. Great work putting this together! @multiphaseCFD & @isschoch can you both touch-base with @AmintorDusko to help check the MacOS wheels locally when you have time?

Will try it and get back to you soon @AmintorDusko @mlxd

File: .github/workflows/wheel_linux_aarch64.yml
Comment (isschoch) [reply .github/workflows/wheel_linux_aarch64.yml:3]: Pip install of whl files worked for me @mlxd and @AmintorDusko on m1

File: .github/workflows/wheel_macos_arm64.yml
Comment (AmintorDusko) [reply .github/workflows/wheel_macos_arm64.yml:28]: Hi @multiphaseCFD, could you please present a log of the tests so we can analyze the failed testing cases?

**Review by mlxd — APPROVED** [review]

**Review by multiphaseCFD — APPROVED** [review]
Nice work! Thanks!

File: .github/workflows/wheel_linux_aarch64.yml
Comment (isschoch) [inline .github/workflows/wheel_linux_aarch64.yml:5]: Typo also appears throughout document I believe.
```suggestion
# **Why we have it**: To build wheels for pennylane-lightning installation.
```

**Review by isschoch — APPROVED** [review]
Nice PR 🥇 . Only found a small type in the comments, not really important but thought I would mention it

File: .github/workflows/wheel_linux_aarch64.yml
Comment (AmintorDusko) [reply .github/workflows/wheel_linux_aarch64.yml:5]: Thank you for mentioning it. I will look into this.

**Review by isschoch — APPROVED** [review]

**Pull request merged**
